### PR TITLE
PROD-84: change phone number validation for user model

### PR DESCRIPTION
**Pull request by shevchukm**
# Reviewers

- [x] @vikabelous
- [x] @okohuch
- [x] @anastasiabalaban
- [x] @Pavliv

# Jira task
https://ssu-jira.softserveinc.com/browse/PROD-84
# Description
fix phone validations for user model, rewrite tests
# Checklist:
- [x] New features are covered with new specs
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

**Review by Pavliv — APPROVED** [review]

**Review by anastasiabalaban — APPROVED** [review]

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -82,16 +82,22 @@
...
   context 'when phone_number is too short' do
     it 'is invalid' do
-      user.phone_number = '+234345'
+      user.phone_number = '+7588108'
       expect(user).to_not be_valid
     end
   end
 
+  context 'when phone_number is correct' do
+    it 'is invalid' do
```
Comment (viktoriia-bilous) [inline spec/models/user_spec.rb]: `it 'is valid'` I suppose

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -82,16 +82,22 @@
...
     it 'is invalid' do
-      user.phone_number = '+234345'
+      user.phone_number = '+7588108'
       expect(user).to_not be_valid
     end
   end
 
+  context 'when phone_number is correct' do
+    it 'is valid' do
+      expect(user).to be_valid
```
Comment (okohuch) [inline spec/models/user_spec.rb]: add shoulda and rewrite tests

**Review by okohuch — CHANGES_REQUESTED** [review]
add shoulda and rewrite tests

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -3,95 +3,19 @@
...
-  end
+  it { is_expected.to validate_presence_of(:surname) }
 
-  context 'when surname field is empty' do
-    it 'is invalid' do
-      user.surname = ''
-      expect(user).to_not be_valid
-    end
-  end
+  it { is_expected.to_not validate_numericality_of(:surname) }
```
Comment (okohuch) [inline spec/models/user_spec.rb]: no need to validate that what we did not add

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -3,95 +3,19 @@
...
-  end
+  it { is_expected.to validate_presence_of(:name) }
 
-  context 'when name field has number' do
-    it 'is invalid' do
-      user.name = 'B1b'
-      expect(user).to_not be_valid
-    end
-  end
+  it { is_expected.to_not validate_numericality_of(:name) }
```
Comment (okohuch) [inline spec/models/user_spec.rb]: no need to validate that what we did not add

File: spec/models/user_spec.rb (hunk 3 above)
Comment (shevchukm) [reply spec/models/user_spec.rb]: @okohuch  done

File: db/migrate/20180418152139_delete_phone_number.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class DeletePhoneNumber < ActiveRecord::Migration[5.1]
+  def change
+    remove_column :users, :phone_number
```
Comment (viktoriia-bilous) [inline db/migrate/20180418152139_delete_phone_number.rb]: Provide the column type to be able to revert this migration
http://guides.rubyonrails.org/active_record_migrations.html#using-the-change-method

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -3,96 +3,20 @@
 RSpec.describe(User, type: :model) do
   let(:user) { build(:user) }
 
-  context 'when name field is empty' do
-    it 'is invalid' do
-      user.name = ''
-      expect(user).to_not be_valid
-    end
+  context 'when user name is empty' do
```
Comment (viktoriia-bilous) [inline spec/models/user_spec.rb]: I would write all test cases under the single `describe 'validations'` block without contexts, they don't add much sense
@okohuch what do you think?

Comment (okohuch) [reply spec/models/user_spec.rb]: @vikabelous I agree

Comment (okohuch) [reply spec/models/user_spec.rb]: all shoulda's functions are self-explanatory.

Comment (okohuch) [reply spec/models/user_spec.rb]: @shevchukm please do changes and share it with the teammates.

**Review by okohuch — APPROVED** [review]

File: spec/models/user_spec.rb (hunk 6 above)
Comment (shevchukm) [reply spec/models/user_spec.rb]: @vikabelous , @okohuch    done

**Review by viktoriia-bilous — APPROVED** [review]

**Pull request merged**